assignment4/disjointset.py

```diff
@@ -0,0 +1,37 @@
+class DisjointSet:
```

Comment:
Given this is a generic implementation of disjoint sets, it would be nice if it had detailed comments.

When defining a generic module, comments of the API often mean that others (or your future self) might be able to use it later on.